### PR TITLE
KOGITO-1269 add businesskey to process instance list

### PR DESCRIPTION
**Pull request by AjayJagan**
- added search bar for businesskey wise search

### CODE REVIEW

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 1]
```diff
@@ -51,11 +51,12 @@ interface IProcessInstance {
   addons: string[];
   endpoint: string;
   error: IProcessInstanceError;
+  businessKey: string;
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: @AjayJagan could we replace _IProcessInstance_ by the _ProcessInstance_ TypeScript type?

File: packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx [hunk 2]
```diff
@@ -149,6 +179,24 @@ const DataToolbarComponent: React.FC<IOwnProps> = ({
...
+          chips={filters.businessKey}
+          deleteChip={onDelete}
+          categoryName="Business key"
+        >
+          <InputGroup>
+            <TextInput
+              name="textInput1"
+              id="textInput1"
+              type="search"
+              aria-label="search input example"
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx]:
```suggestion
              aria-label="buness key"
```

File: packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx [hunk 3]
```diff
@@ -149,6 +179,24 @@ const DataToolbarComponent: React.FC<IOwnProps> = ({
...
         </DataToolbarFilter>
+        <DataToolbarFilter
+          chips={filters.businessKey}
+          deleteChip={onDelete}
+          categoryName="Business key"
+        >
+          <InputGroup>
+            <TextInput
+              name="textInput1"
+              id="textInput1"
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx]:
```suggestion
              id="businesskey"
```

File: packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx [hunk 4]
```diff
@@ -149,6 +179,24 @@ const DataToolbarComponent: React.FC<IOwnProps> = ({
...
           </Select>
         </DataToolbarFilter>
+        <DataToolbarFilter
+          chips={filters.businessKey}
+          deleteChip={onDelete}
+          categoryName="Business key"
+        >
+          <InputGroup>
+            <TextInput
+              name="textInput1"
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx]:
```suggestion
              name="businesskey"
```

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx (hunk 1 above)
Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: done

File: packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx (hunk 2 above)
Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx]: done

File: packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx (hunk 3 above)
Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx]: done

File: packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx (hunk 4 above)
Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DataToolbarComponent/DataToolbarComponent.tsx]: done

**Review by cristianonicolai — APPROVED** [review]
looks good.

@AjayJagan @srambach would be ok if we apply the business key value on enter? like enter value, hit enter then results are reloaded.

**Comment by AjayJagan** [comment]
> looks good.
> 
> @AjayJagan @srambach would be ok if we apply the business key value on enter? like enter value, hit enter then results are reloaded.

I can implement that if you like to :)

**Comment by AjayJagan** [comment]
> > looks good.
> > @AjayJagan @srambach would be ok if we apply the business key value on enter? like enter value, hit enter then results are reloaded.
> 
> I can implement that if you like to :)

I have added it. I can push it if you would like to

**Comment by cristianonicolai** [comment]
@AjayJagan +1 from me

**Comment by AjayJagan** [comment]
> @AjayJagan +1 from me

done :+1:

**Comment by cristianonicolai** [comment]
@AjayJagan looks nice, thanks.

**Comment by srambach** [comment]
A few comments:
- Agree, Enter within the filter text box should apply the filter 👍 
- I can't type a capital letter in the box - is it being transformed so it's case insensitive or what exactly is going on there? 
- Change the placeholder in the box to "Filter by business key"
- In the display, the process name needs to be distinct from the business key. Right now, they appear to just be one that happens to have a hyphen in it. Either it should be its own column, or we need to use a different font style. I think this should be applied throughout, in the common component that @cristianonicolai was talking about. I don't have a specific recommendation at the moment.

**Comment by srambach** [comment]
One more - I think the button `Apply Filter` should be sentence case `Apply filter`

**Comment by cristianonicolai** [comment]
@srambach correct, that should be the common component I mentioned.

**Comment by AjayJagan** [comment]
> @srambach correct, that should be the common component I mentioned.

So I will design the components with the fonts that you recommend

**Comment by AjayJagan** [comment]
> * Filter by business key

 - enter is added 
 - Capital letters can be typed now
 - added Filter by business key text
 -  I guess the separate component for displaying processId-(Businesskey||id) can be done in a separate pr. :)

**Review by cristianonicolai — APPROVED** [review]

**Comment by srambach** [comment]
The filter seems to be working well now, and on the Enter key is good.
With the hyphen, the business key looks like one long string instead of 2 separate pieces of data. For now how about if we wrap a Badge around the business key? And then I'm ok with merging this. 
Long term we should have a design Jira and then a Jira to make it be its own component so that it can be reused everywhere consistently. 
with the business key as `<Badge>{processInstanceData.businessKey}</Badge>`:
image

**Comment by AjayJagan** [comment]
> The filter seems to be working well now, and on the Enter key is good.
> With the hyphen, the business key looks like one long string instead of 2 separate pieces of data. For now how about if we wrap a Badge around the business key? And then I'm ok with merging this.
> Long term we should have a design Jira and then a Jira to make it be its own component so that it can be reused everywhere consistently.
> with the business key as `<Badge>{processInstanceData.businessKey}</Badge>`:
> image

@srambach - yeah this looks good. we would be having 2 cases.
1) if business key is present we can use this.
2) what would be the design if we show the spliced 'id'.

Maybe we go with this JIRA as it is and we can add the design changes in this JIRA : https://issues.redhat.com/browse/KOGITO-1447. wdyt @cristianonicolai

**Comment by cristianonicolai** [comment]
@srambach the bagde idea looks nice and I agree with @AjayJagan we can do that in the followup JIRA.

**Review by srambach — APPROVED** [review]
Approved but we need to prioritize the changes to displaying the instance name, ID, and business key in Kogito-1447

**Comment by Sara4994** [comment]
@AjayJagan  I have selected Active and Aborted states from dropdown, which gives me 3 process instances and i see a business key "Hello World", when i search that key I get "No results found" . Also i see the text transformed to lowercase in the business key chip which is not giving us correct result.

@cristianonicolai  will business key contain spaces between like "Hello World"? 

Screenshot from 2020-03-19 21-43-06
Screenshot from 2020-03-19 21-39-37

**Comment by AjayJagan** [comment]
> @AjayJagan I have selected Active and Aborted states from dropdown, which gives me 3 process instances and i see a business key "Hello World", when i search that key I get "No results found" . Also i see the text transformed to lowercase in the business key chip which is not giving us correct result
> 
> Screenshot from 2020-03-19 21-43-06
> Screenshot from 2020-03-19 21-39-37

I will look at the case issue , but there is no business Key as Hello World. I am not sure why this hello world is comming up. let me look at it.

**Comment by cristianonicolai** [comment]
@Sara4994 @AjayJagan yep, business key can have empty string/space.

**Comment by AjayJagan** [comment]
> > @AjayJagan I have selected Active and Aborted states from dropdown, which gives me 3 process instances and i see a business key "Hello World", when i search that key I get "No results found" . Also i see the text transformed to lowercase in the business key chip which is not giving us correct result
> > Screenshot from 2020-03-19 21-43-06
> > Screenshot from 2020-03-19 21-39-37
> 
> I will look at the case issue , but there is no business Key as Hello World. I am not sure why this hello world is comming up. let me look at it.

@Sara4994 -There are 2 things here
1) I was looking the mock server and for those process without the business key attribute(when it doesnot have a value or even null is given), it automatically creates this businessKey:'Hello World'. do you have any idea why this happens?
2) The css written for transforming the state(capitalize) is also affecting the chip group for the business key search word . @srambach - can you please have a look at the css part

**Comment by Sara4994** [comment]
> > > @AjayJagan I have selected Active and Aborted states from dropdown, which gives me 3 process instances and i see a business key "Hello World", when i search that key I get "No results found" . Also i see the text transformed to lowercase in the business key chip which is not giving us correct result
> > > Screenshot from 2020-03-19 21-43-06
> > > Screenshot from 2020-03-19 21-39-37
> > 
> > 
> > I will look at the case issue , but there is no business Key as Hello World. I am not sure why this hello world is comming up. let me look at it.
> 
> @Sara4994 -There are 2 things here
> 
> 1. I was looking the mock server and for those process without the business key attribute(when it doesnot have a value or even null is given), it automatically creates this businessKey:'Hello World'. do you have any idea why this happens?
> 2. The css written for transforming the state(capitalize) is also affecting the chip group for the business key search word . @srambach - can you please have a look at the css part

@AjayJagan  to exclude business key chips from text transformation and to target only state chips, you can add below block in the css file. Now i could see chip text same as what i type in the text box, but still it gives me "No results found" when i search it.

.kogito-management-console__state-dropdown-list .pf-c-data-toolbar__item:first-child .pf-c-chip__text::first-letter {
  text-transform: uppercase;
}
.kogito-management-console__state-dropdown-list .pf-c-data-toolbar__item:first-child .pf-c-chip__text {
  text-transform: lowercase;
}

.kogito-management-console__state-dropdown-list .pf-c-check__label::first-letter {
  text-transform: uppercase;
}

.kogito-management-console__state-dropdown-list .pf-c-check__label {
  text-transform: lowercase;
}
Screenshot from 2020-03-20 13-07-13

**Comment by AjayJagan** [comment]
> > > > @AjayJagan I have selected Active and Aborted states from dropdown, which gives me 3 process instances and i see a business key "Hello World", when i search that key I get "No results found" . Also i see the text transformed to lowercase in the business key chip which is not giving us correct result
> > > > Screenshot from 2020-03-19 21-43-06
> > > > Screenshot from 2020-03-19 21-39-37
> > > 
> > > 
> > > I will look at the case issue , but there is no business Key as Hello World. I am not sure why this hello world is comming up. let me look at it.
> > 
> > 
> > @Sara4994 -There are 2 things here
> > 
> > 1. I was looking the mock server and for those process without the business key attribute(when it doesnot have a value or even null is given), it automatically creates this businessKey:'Hello World'. do you have any idea why this happens?
> > 2. The css written for transforming the state(capitalize) is also affecting the chip group for the business key search word . @srambach - can you please have a look at the css part
> 
> @AjayJagan to exclude business key chips from text transformation and to target only state chips, you can add below block in the css file. Now i could see chip text same as what i type in the text box, but still it gives me "No results found" when i search it.
> 
> .kogito-management-console__state-dropdown-list .pf-c-data-toolbar__item:first-child .pf-c-chip__text::first-letter {
> text-transform: uppercase;
> }
> .kogito-management-console__state-dropdown-list .pf-c-data-toolbar__item:first-child .pf-c-chip__text {
> text-transform: lowercase;
> }
> 
> .kogito-management-console__state-dropdown-list .pf-c-check__label::first-letter {
> text-transform: uppercase;
> }
> 
> .kogito-management-console__state-dropdown-list .pf-c-check__label {
> text-transform: lowercase;
> }
> Screenshot from 2020-03-20 13-07-13

@Sara4994 - yes it would give no results because we donot have a business key with hello world in the mock server

**Comment by Sara4994** [comment]
Looks good now. :)

**Review by Sara4994 — APPROVED** [review]

**Pull request merged**